### DISCUSSION TITLE
Allow placing factories by Game Script freely

### DISCUSSION THREAD

**Discussion by npabisz** [OpenTTD/OpenTTD#10900] [Features]
### Version of OpenTTD

13.1

### Expected result

I'm able to place factory through GS wherever I want (as long as it is one per town).

### Actual result

Climates like sub-tropical or arctic has limitation on some factories which cannot be placed in specific areas, eg. snow, desert or Oil Factory which cannot be placed if not close to map edge. 

I think this should be not applicable to Game Scripts which should have free hand on such things. Right now any climate, but temperate is useless for City Builder game modes which would require different types of cargo delivered to town.

### Steps to reproduce

1. Pick sub-tropical climate
2. Fund Factory by GS on desert

**Comment by Eddi-z** [comment]
These placement restrictions are traditionally controlled by NewGRFs, it would be a bit weird to interfere in their territory like this.

**Comment by npabisz** [comment]
I mean I'm not sure why industries which are not directly producing resources have these restrictions in first place. I can understand Rubber Plantation to not being able to be placed on desert, but Factory? Like why Factory can't be placed on desert while Food Processing Plant can? Why Oil Factory has to be placed on map edges? Maybe its time to redefine these restrictions to be more logical. Like in real world they don't care where factories are built. Mines, farms, plantations, etc. are different story.

**Reply by TrueBrain** [reply]
As this isn't a bug (it is fully working as intended), I moved the ticket to a discussion, so you can talk over with NewGRF authors why they add restrictions to industries.

But as @Eddi-z explained, NewGRFs (and vanilla) define how industries can be placed; currently a GS cannot overrule that decision, and this is by design. But as that goes with anything OpenTTD, everything is up for discussion, and discussions are the perfect place for that :)

**Reply by glx22** [reply]
Area restrictions are not the worst, some NewGRFs have heavily complex landscape requirements.
